pipelinewise/cli/pipelinewise.py

```diff
@@ -1729,6 +1731,42 @@ def _run_post_import_tap_checks(
 
         return errors
 
+    def _cleanup_tap_state_file(self) -> None:
+        tables = self.args.tables
+        state_file = self.tap['files']['state']
+        if tables:
+            self._clean_tables_from_bookmarks_in_state_file(state_file, tables)
+        else:
+            try:
+                os.remove(state_file)
+            except FileNotFoundError:
+                pass
+
+    @staticmethod
+    def _clean_tables_from_bookmarks_in_state_file(state_file_to_clean: str, tables: str) -> None:
+        try:
+            with open(state_file_to_clean, 'r+', encoding='UTF-8') as state_file:
+                state_data = json.load(state_file)
+                bookmarks = state_data.get('bookmarks')
+                list_of_tables = tables.split(',')
+                if bookmarks:
+                    cleanedup_bookmarks = {
+                        bookmark_name: bookmarks[bookmark_name]
+                        for bookmark_name in bookmarks
+                        if not any(table_name in bookmark_name for table_name in list_of_tables)
+                    }
+
+                    state_data['bookmarks'] = cleanedup_bookmarks
+
+                state_file.seek(0)
+                json.dump(state_data, state_file)
+                state_file.truncate()
```

Comment:
why do you need this?

Reply:
this overwrites the cleaned up json in the file:
`state_file.seek(0)` : moving the pointer to the beginning of the file.
`json.dump(state_data, state_file)` : dumping the cleaned up `state_data` to the file.
`state_file.truncate()`: removing the remained bytes of the previous contenx

i.e:
 
 original content of the file = `footbar`
 new data = `test`
 
 if not seek(0) then new content of the file will be : `foobartest`
 if use seek(0) but not truncate, the the contnt of the file will be: `tester`

Reply:
got it, thanks!